### ISSUE TITLE
feathers-mongoose@3.6.0 break custom model id

### ISSUE THREAD

**Issue by loicbisiere** [feathersjs/feathers-mongoose#151]
### Steps to reproduce

Create a simple service with a custom model id :  
```
"use strict";

const service = require("feathers-mongoose");
const orders = require("./orders-model");
const hooks = require("./hooks");

module.exports = function() {
  const app = this;

  const options = {
    Model: orders,
    id: "oId"
  };

  // Initialize our service with any options it requires
  app.use("/orders", service(options));

  // Get our initialize service to that we can bind hooks
  const ordersService = app.service("/orders");

  // Set up our before hooks
  ordersService.before(hooks.before);

  // Set up our after hooks
  ordersService.after(hooks.after);
};
```

orders-models.js :  
```
"use strict";

// orders-model.js - A mongoose model
// 
// See http://mongoosejs.com/docs/models.html
// for more of what you can do here.

const mongoose = require("mongoose");
const Schema = mongoose.Schema;
const Mixed = Schema.Types.Mixed;

const ordersSchema = new Schema({
  oId: {
    type: String,
    required: true,
    trim: true,
    index: {
      unique: true
    }
  },
  amount: {
    type: Number,
    required: true,
    "default": 0
  },
  ...
  state: {
    type: String,
    "default": "new"
  }
});

const ordersModel = mongoose.model("orders", ordersSchema);

module.exports = ordersModel;
```

Create a "before" hook to initialize the "oId" on create :  
```
exports.create = function() {
  return function(hook) {
    var randomId = Math.random().toString(36).substr(3, 2);
    hook.data.oId = "PO"+"-"+randomId+"-"+moment().unix();
    hook.data.state = "new";
  };
};
```

### Expected behavior
Send a Post request to create a new "Order" without oId. The Order will be created with a oId generated by the hook. 

### Actual behavior
Mongoose validator send a missing error about the oId.

### System configuration

Error happen since I updated feathers-mongoose@3.6.0.
No problem with feathers-mongoose@3.5.3

**Comment by daffl** [comment]
Can you post the `order-model` as well?

**Comment by daffl** [comment]
I bet the problem is https://github.com/feathersjs/feathers-mongoose/commit/1222843093f738531fabc1002a5f67b76c45048b#diff-480db08609ef6a51bf20f049cc9bda48R137

**Comment by loicbisiere** [comment]
I updated my first post with the orders-model.
Thank you.

**Comment by daffl** [comment]
This has been fixed in v3.6.1
